dependencies.repos

```diff
@@ -5,7 +5,7 @@
 - git:
     local-name: boost_plugin_loader
     uri: https://github.com/tesseract-robotics/boost_plugin_loader.git
-    version: 0.3.0
+    version: main
```

Comment:
Per request in https://github.com/tesseract-robotics/tesseract/pull/1191#issuecomment-3164548586, I will say we always prefer to track named versions (tags or commit hashes) over named branches that can become moving targets over time